src/components/ScreenWrapper.js

```diff
@@ -1,147 +0,0 @@
-import _ from 'underscore';
-import React from 'react';
-import PropTypes from 'prop-types';
-import {View} from 'react-native';
-import {SafeAreaInsetsContext} from 'react-native-safe-area-context';
-import {withOnyx} from 'react-native-onyx';
-import styles from '../styles/styles';
-import * as StyleUtils from '../styles/StyleUtils';
-import HeaderGap from './HeaderGap';
-import KeyboardShortcutsModal from './KeyboardShortcutsModal';
-import KeyboardShortcut from '../libs/KeyboardShortcut';
-import onScreenTransitionEnd from '../libs/onScreenTransitionEnd';
-import Navigation from '../libs/Navigation/Navigation';
-import compose from '../libs/compose';
-import ONYXKEYS from '../ONYXKEYS';
-import CONST from '../CONST';
-import withNavigation from './withNavigation';
-
-const propTypes = {
-    /** Array of additional styles to add */
-    style: PropTypes.arrayOf(PropTypes.object),
-
-    /** Returns a function as a child to pass insets to or a node to render without insets */
-    children: PropTypes.oneOfType([
-        PropTypes.node,
-        PropTypes.func,
-    ]).isRequired,
-
-    /** Whether to include padding bottom */
-    includePaddingBottom: PropTypes.bool,
-
-    /** Whether to include padding top */
-    includePaddingTop: PropTypes.bool,
-
-    // Called when navigated Screen's transition is finished.
-    onTransitionEnd: PropTypes.func,
-
-    // react-navigation navigation object available to screen components
-    navigation: PropTypes.shape({
-        // Method to attach listener to Navigation state.
-        addListener: PropTypes.func.isRequired,
-    }),
```

Comment:
NAB: I don't see these propTypes in BaseScreenWrapper. Is withNavigation is somehow applying the navigation prop type. If so, how does that work?

Reply:
Yes, `withNavigation` automatically applies those propTypes. I didn't include them in the propTypes because we don't include them anywhere else we use `withNavigation`, for instance [here](https://github.com/Expensify/App/blob/631994f7b9a974ca3ce92b8405bd1416a9d34623/src/components/SettlementButton.js#L97) with the SettlementButton.

Reply:
Yeah I noticed that the prop types aren't used elsewhere and I'm wondering why that is and if / how they are applied at all. I'll ask in Slack for my own curiosity.